From dcd758480d1c6a61b66b983a900b7ac1e9415c74 Mon Sep 17 00:00:00 2001
From: LIU Hao <lh_mouse@126.com>
Date: Fri, 6 Jan 2023 22:16:45 +0800
Subject: [PATCH 1/2] gmp: Do not declare functions as `dllimport`

GNU toolchains do not require `dllimport` when calling functions from
DLLs. Having `dllimport` by default causes static linking to fail,
observed when building GDB.

Signed-off-by: LIU Hao <lh_mouse@126.com>

Source:

https://github.com/msys2/MINGW-packages/commit/00f2f6c12a969e729c1fc111f5f11e071602677e
---
 gmp-h.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gmp-h.in b/gmp-h.in
index 8598e97..340dd9e 100644
--- a/gmp-h.in
+++ b/gmp-h.in
@@ -102,7 +102,7 @@ see https://www.gnu.org/licenses/.  */
 
 #if defined (__GNUC__)
 #define __GMP_DECLSPEC_EXPORT  __declspec(__dllexport__)
-#define __GMP_DECLSPEC_IMPORT  __declspec(__dllimport__)
+#define __GMP_DECLSPEC_IMPORT
 #endif
 #if defined (_MSC_VER) || defined (__BORLANDC__)
 #define __GMP_DECLSPEC_EXPORT  __declspec(dllexport)
-- 
2.44.0.windows.1

